Sources/BraintreeDropIn/Models/BTDropInResult.m

```diff
@@ -11,30 +11,67 @@
 #import <BraintreeCore/BraintreeCore.h>
 #endif
 
+NSString *const BTDropInResultErrorDomain = @"com.braintreepayments.BTDropInResultErrorDomain";
+
 @implementation BTDropInResult
 
 #pragma mark - Prefetch BTDropInResult
 
-+ (void)fetchDropInResultForAuthorization:(NSString *)authorization handler:(BTDropInResultFetchHandler)handler {
-    BTUIKPaymentOptionType lastSelectedPaymentOptionType = [[NSUserDefaults standardUserDefaults] integerForKey:@"BT_dropInLastSelectedPaymentMethodType"];
-    __block BTAPIClient *apiClient = [[BTAPIClient alloc] initWithAuthorization:authorization sendAnalyticsEvent:NO];
+static NSUserDefaults *_userDefaults = nil;
+
++ (NSUserDefaults *)userDefaults {
+    if (_userDefaults == nil) {
+        _userDefaults = [NSUserDefaults standardUserDefaults];
+    }
+    return _userDefaults;
+}
+
++ (void)setUserDefaults:(NSUserDefaults *)userDefaults {
+    if (userDefaults != _userDefaults) {
+        _userDefaults = userDefaults;
+    }
+}
+
++ (void)mostRecentPaymentMethodForClientToken:(NSString *)clientToken
+                                   completion:(void (^)(BTDropInResult *result, NSError *error))completion {
+    BTAPIClient *apiClient = [[BTAPIClient alloc] initWithAuthorization:clientToken sendAnalyticsEvent:NO];
```

Comment:
Idea - is it worth doing a client side validation here that the `clientToken` is actually a clientToken and returning back a more specific error?

Reply:
That error will bubble up from the [base SDK](https://github.com/braintree/braintree_ios/blob/7ccda0f255230b8d427258766cb8041ba5ead5a4/Sources/BraintreeCore/BTAPIClient.m#L264).